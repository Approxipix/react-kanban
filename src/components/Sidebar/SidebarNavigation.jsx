import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import SidebarNavigationList from './SidebarNavigationList';
import styled from 'styled-components';

const Title = styled.h3`
  font-weight: 600;
`;

const List = styled.ul``;
List.displayName = 'List';

const Item = styled.li`
  margin-bottom: .5rem;
  padding: .3rem .5rem;
  border-radius: .2rem;
  transition: background-color .2s ease-in;
  cursor: pointer;
  &:hover {
    background-color: rgba(9,30,66,.13);
  }
`;

const Link = styled.a`
  display: flex;
  align-items: center;
  font-size: .875rem;
  font-weight: 700;
`;
Link.displayName = 'Link';

const Icon = styled.i`
  margin-right: .75rem;
  color: #42526e;
`;

class SidebarNavigation extends Component {
  constructor(props) {
    super(props);
    this.state = {
      navigation: SidebarNavigationList,
    }
  }

  changeView = (e, item) => {
    e.preventDefault();
    this.props.changeView(item.component, item.title)
  };

  render() {
    const { navigation } = this.state;
    return (
      <List>
        {navigation.map((item, index) => (
          <Item key={index} >
            <Link onClick={(e) => this.changeView(e, item)}>
              <Icon>
                <FontAwesomeIcon icon={item.icon} />
              </Icon>
              <Title>
                {item.title}
              </Title>
            </Link>
          </Item>
        ))}
      </List>
    );
  }
}

SidebarNavigation.defaultProps = {
  changeView: () => {},
};

SidebarNavigation.propTypes = {
  changeView: PropTypes.func.isRequired,
};

export default SidebarNavigation;
