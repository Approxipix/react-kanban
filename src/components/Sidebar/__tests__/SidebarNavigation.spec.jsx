import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import SidebarNavigation from '../SidebarNavigation';

describe('<SidebarNavigation>', () => {
  let component;
  let mockChangeView = jest.fn();

  beforeEach(() => {
    const state = {
      navigation: [
        { title: 'Change Background', component: 'BoardBackgroundEdit', icon: 'fill-drip' },
        { title: 'Delete Board', component: 'BoardDelete', icon: 'trash' },
      ]
    };

    component = shallow(<SidebarNavigation changeView={mockChangeView} />);
    component.setState({ ...state });
  });

  it('should renders without crashing', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render a nav link for each item from state', () => {
    expect(component.find('Link').length).toEqual(2)
  });

  it("should call changeView after click on nav item", () => {
    component.find('Link').first().simulate('click', {
      preventDefault: () => {}
    });

    expect(mockChangeView).toHaveBeenCalledTimes(1);
    expect(mockChangeView).toHaveBeenCalledWith('BoardBackgroundEdit', 'Change Background');
  });
});
