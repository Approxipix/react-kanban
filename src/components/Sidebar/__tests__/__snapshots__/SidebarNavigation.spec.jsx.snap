// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarNavigation> should renders without crashing 1`] = `
<List>
  <styled.li
    key="0"
  >
    <Link
      onClick={[Function]}
    >
      <styled.i>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="fill-drip"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </styled.i>
      <styled.h3>
        Change Background
      </styled.h3>
    </Link>
  </styled.li>
  <styled.li
    key="1"
  >
    <Link
      onClick={[Function]}
    >
      <styled.i>
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon="trash"
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </styled.i>
      <styled.h3>
        Delete Board
      </styled.h3>
    </Link>
  </styled.li>
</List>
`;
